bundle/src/main/java/com/adobe/acs/commons/indesign/dynamicdeckdynamo/services/impl/DynamicDeckServiceImpl.java

```diff
@@ -0,0 +1,224 @@
+        }
+
+        InputStream templateXmlInputStream = DynamicDeckUtils.getAssetInputStreamByPath(resourceResolver, annotatedXmlPath);
+        if (null == templateXmlInputStream) {
+            throw new DynamicDeckDynamoException("InDesign Template XML is null, hence exiting the deck generation process");
+        }
+
+        List<XMLResourceIterator> assetItrList = new ArrayList<>();
+        assetItrList.add(new XMLResourceIterator(DynamicDeckDynamoConstants.XML_SECTION_TYPE_GENERIC, assetResourceList.listIterator()));
+
+        String processedXmlPath = xmlGeneratorService.generateInddXML(templateXmlInputStream, assetItrList,
+                resourceResolver.getResource(masterAssetPath), damAsset.adaptTo(Resource.class), resourceResolver, inddImageList);
+
+        StringBuilder exportFormats = DynamicDeckUtils.addExportFormat(damAsset, allFormats);
+        StringBuilder imagePaths = DynamicDeckUtils.getImagePaths(inddImageList, configurationService.getPlaceholderImagePath());
+
+        addIdsScriptArgs(inddTemplatePath, damAsset.adaptTo(Resource.class), idspScriptArgs, imagePaths, processedXmlPath, exportFormats);
+        DynamicDeckUtils.commit(resourceResolver);
+
+        return damAsset.getPath();
+    }
+
+    /**
+     * This method fetch asset list from collection/smart-collection.
+     * <p>
+     * TODO Planning to create service account and remove the dependency of resourceResolver from this service
+     *
+     * @param collectionPath
+     * @param resourceResolver
+     * @return
+     */
+    @Override
+    public List<Resource> fetchAssetListFromCollection(String collectionPath, ResourceResolver resourceResolver) throws DynamicDeckDynamoException {
+        List<Resource> arrayList = new ArrayList<>();
+        if (StringUtils.isBlank(collectionPath)) {
+            throw new DynamicDeckDynamoException("Resource Path is Null/Empty ");
+        }
+        Resource collectionResource = resourceResolver.getResource(collectionPath);
+        if (collectionResource == null) {
+            throw new DynamicDeckDynamoException("No Resource exists at Supplied Path :" + collectionPath);
+        }
+
+        if (DamUtil.isSmartCollection(collectionResource)) {
+            return DynamicDeckUtils.fetchSmartCollectionResourceList(collectionResource);
+        }
+
+        Iterator<Asset> assets = DamUtil.getAssets(collectionResource);
+
+        assets.forEachRemaining(eachAsset -> {
+            Resource eachAssetResource = eachAsset.adaptTo(Resource.class);
+            if (null != eachAssetResource) {
+                arrayList.add(eachAsset.adaptTo(Resource.class));
+            }
+        });
+        return arrayList;
+    }
+
+    @Override
+    public List<Resource> fetchAssetListFromQuery(String queryString, ResourceResolver resourceResolver) {
+        Iterator<Resource> assetResourceIterator = resourceResolver.findResources(queryString, Query.JCR_SQL2);
```

Comment:
No need to do this now, but ACS Commons has the QueryHelper OSGi Service that makes it really easy to pass in all sorts of queries, including QueryBuilder syntax. If you pass in a "language" parameter (which currently defaults to Query.JCR_SQL2, it would set this up nicely for using that.

Reply:
Sure, noted, we'll incorporate it in a future enhancement.